src/app/usePayment.tsx

```diff
@@ -252,18 +252,51 @@ export const PaymentProvider: FC<
           return "";
         } else if (oldOrderData?.googlePayData) {
           //@ts-ignore
-          const { status } = await paypal.Googlepay().confirmOrder({
-            orderId: id,
+          const confirmOrderResult = await paypal.Googlepay().confirmOrder({
+            orderId: orderData.id,
             paymentMethodData:
               oldOrderData.googlePayData.paymentData.paymentMethodData,
           });
+          const { status } = confirmOrderResult;
           if (status === "APPROVED") {
             handleOnApprove({ orderID: orderData.id }).then(() =>
               onSuccess(orderData)
             );
           } else if (status === "PAYER_ACTION_REQUIRED") {
```

Comment:
We need to check if the Payment Method is GooglePay and status === "PAYER_ACTION_REQUIRED", then do next